src/hardhat/provider/index.ts

```diff
@@ -0,0 +1,30 @@
+import { ethers } from "ethers";
+import { EthereumProvider } from "hardhat/types";
+
+export class EthersProviderWrapper extends ethers.providers.JsonRpcProvider {
```

Comment:
why do we need this wrapper?
Is it to stay compatible with the max rpc call limit error handling (relying on `debug` events)?

Reply:
It is just a hack to use ethers' provider class in the plugin rather than using hardhat's provider and doing `provider.send(JSON-rpc method)`